src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/RemoteTypes/RemoteSourceRepositoryProvider.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Threading;
+using NuGet.Configuration;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.VisualStudio;
+
+namespace NuGet.PackageManagement.VisualStudio.RemoteTypes
+{
+    [Export(typeof(IRemoteSourceRepositoryProvider))]
+    internal sealed class RemoteSourceRepositoryProvider : IRemoteSourceRepositoryProvider
+    {
+        public IPackageSourceProvider PackageSourceProvider => new RemotePackageSourceProvider();
+
+        public SourceRepository CreateRepository(PackageSource source)
+        {
+            return NuGetUIThreadHelper.JoinableTaskFactory.Run(async () =>
```

Comment:
Same feedback: Please avoid JTF.Run around brokered service calls. Allow the asynchronony to propagate all the way up to your top-level API, then update your UX to be async.